compiler/src/dotty/tools/dotc/interactive/Interactive.scala

```diff
@@ -413,4 +446,43 @@ object Interactive {
   /** The first tree in the path that is a definition. */
   def enclosingDefinitionInPath(path: List[Tree])(implicit ctx: Context): Tree =
     path.find(_.isInstanceOf[DefTree]).getOrElse(EmptyTree)
+
+  /** All the symbols that are imported by import statement `imp`, if it matches
+   *  the predicate `selectorPredicate`.
+   *
+   *  @param imp The import statement to analyze
+   *  @param selectorPredicate A test to find the selector to use.
+   *  @return The symbols imported.
+   */
+  def importedSymbols(imp: tpd.Import,
+                      selectorPredicate: untpd.Tree => Boolean = util.common.alwaysTrue)
+                     (implicit ctx: Context): List[Symbol] = {
+    def lookup0(name: Name): Symbol = imp.expr.tpe.member(name).symbol
+    def lookup(name: Name): List[Symbol] = {
+      lookup0(name.toTermName) ::
+        lookup0(name.toTypeName) ::
+        lookup0(name.moduleClassName) ::
+        lookup0(name.sourceModuleName) :: Nil
+    }
+
+    val symbols = imp.selectors.find(selectorPredicate) match {
+      case Some(id: untpd.Ident) =>
+        lookup(id.name)
+      case Some(Thicket((id: untpd.Ident) :: (_: untpd.Ident) :: Nil)) =>
+        lookup(id.name)
+      case _ => Nil
+    }
+
+    symbols.filter(_.exists)
+  }
+
+  /** Used to represent a renaming import `{foo => bar}`.
+   *  We need this because the name of the tree must be the new name, but the
+   *  denotation must be that of the importee.
+   */
+  private case class RenameTree(name: Name, underlying: Tree) extends NameTree {
```

Comment:
Do we really need to create a new AST node if the only thing we do on it is traverse it once and not store it anywhere afterwards?

Reply:
The goal is to have a `NameTree` (we need a `NameTree` because that is what `namedTrees` is looking for), such that:
 - Its `denot` is the denotation of the thing that is actually imported, so that comparing symbols works as expected
 - Its `name` is the name of the symbol after renaming, so that `namePos` is correct.

The tree may be stored and returned by `namedTrees`.

Should I try to make the doc comment clearer, or do you have a better idea to achieve the same result?